### PR TITLE
Update/Cancel/Approve holidays

### PR DESCRIPTION
**Pull request by paulbrownlee**

### CODE REVIEW

File: BackEnd/src/main/java/com/unosquare/admin_core/back_end/ViewModels/SaveWorkingFromHomeViewModel.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+public class SaveWorkingFromHomeViewModel {
+
+    @NotBlank
+    private List<DateViewModel> dates;
+
+    @NotBlank
+    private int employeeId;
+
+    @NotBlank
+    private int eventTypeId;
```
Comment (BrownMark) [inline BackEnd/src/main/java/com/unosquare/admin_core/back_end/ViewModels/SaveWorkingFromHomeViewModel.java:18]: This is inferred by the type.

File: BackEnd/src/main/java/com/unosquare/admin_core/back_end/ViewModels/UpdateHolidayViewModel.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+@Data
+public class UpdateHolidayViewModel {
+    @NotBlank
+    private List<DateViewModel> dates;
+
+    @NotBlank
+    private int employeeId;
+
+    @NotBlank
+    private int eventTypeId;
```
Comment (BrownMark) [inline BackEnd/src/main/java/com/unosquare/admin_core/back_end/ViewModels/UpdateHolidayViewModel.java]: This is inferred by the type.

File: BackEnd/src/main/java/com/unosquare/admin_core/back_end/ViewModels/UpdateHolidayViewModel.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    @NotBlank
+    private int eventStatusId;
```
Comment (BrownMark) [inline BackEnd/src/main/java/com/unosquare/admin_core/back_end/ViewModels/UpdateHolidayViewModel.java]: We'll make specific calls for the status, let's not specify like this.

File: BackEnd/src/main/java/com/unosquare/admin_core/back_end/controller/HolidayController.java [hunk 4]
```diff
@@ -90,14 +89,41 @@ public ResponseEntity createHoliday(@RequestBody SaveHolidayViewModel saveHolida
...
         EventDTO event = modelMapper.map(saveHolidayViewModel, EventDTO.class);
         eventService.saveEvents(event);
+    }*/
+
+    @PutMapping(value = "/", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public void updateHoliday(@RequestBody UpdateHolidayViewModel updateHolidayViewModel) {
+        for(DateViewModel date : updateHolidayViewModel.getDates()){
+            EventDTO event = modelMapper.map(date, EventDTO.class);
+            event.setEventId(updateHolidayViewModel.getHolidayId());
```
Comment (BrownMark) [inline BackEnd/src/main/java/com/unosquare/admin_core/back_end/controller/HolidayController.java]: Should this not happen in the mapping?

File: BackEnd/src/main/java/com/unosquare/admin_core/back_end/service/EventService.java [hunk 5]
```diff
@@ -60,20 +60,43 @@ private void save(Event event) {
...
             Event event = modelMapper.map(eventDto,Event.class);
             save(event);
         }
     }
 
+    public  void updateEvent(EventDTO eventDTO){
+        Preconditions.checkNotNull(eventDTO);
+        EventDTO temp = findById(eventDTO.getEventId());
+        Event event = modelMapper.map(temp, Event.class);
+        event.setStartDate(eventDTO.getStartDate());
```
Comment (BrownMark) [inline BackEnd/src/main/java/com/unosquare/admin_core/back_end/service/EventService.java]: Same with mappings.

File: BackEnd/src/main/java/com/unosquare/admin_core/back_end/service/EventService.java [hunk 6]
```diff
@@ -60,20 +60,43 @@ private void save(Event event) {
...
-
         for(EventDTO eventDto : eventDtos) {
             Event event = modelMapper.map(eventDto,Event.class);
             save(event);
         }
     }
 
+    public  void updateEvent(EventDTO eventDTO){
+        Preconditions.checkNotNull(eventDTO);
+        EventDTO temp = findById(eventDTO.getEventId());
```
Comment (BrownMark) [inline BackEnd/src/main/java/com/unosquare/admin_core/back_end/service/EventService.java]: I'd use the reppsitory here.

File: BackEnd/src/main/java/com/unosquare/admin_core/back_end/service/EventService.java [hunk 7]
```diff
@@ -60,20 +60,43 @@ private void save(Event event) {
...
+        Event event = modelMapper.map(temp, Event.class);
+        event.setStartDate(eventDTO.getStartDate());
+        event.setEndDate(eventDTO.getEndDate());
+        save(event);
+    }
+
+    public void approveEvent(EventDTO eventDTO){
+        Preconditions.checkNotNull(eventDTO);
+        EventDTO temp = findById(eventDTO.getEventId());
```
Comment (BrownMark) [inline BackEnd/src/main/java/com/unosquare/admin_core/back_end/service/EventService.java]: I'd use the repository here.

File: BackEnd/src/main/java/com/unosquare/admin_core/back_end/service/EventService.java [hunk 8]
```diff
@@ -60,20 +60,43 @@ private void save(Event event) {
...
+        Event event = modelMapper.map(temp,Event.class);
+        event.getEventStatus().setEventStatusId(EventStatuses.APPROVED.getEventStatusId());
+        save(event);
+    }
+
+    public void cancelEvent(EventDTO eventDTO){
+        Preconditions.checkNotNull(eventDTO);
+        EventDTO temp = findById(eventDTO.getEventId());
```
Comment (BrownMark) [inline BackEnd/src/main/java/com/unosquare/admin_core/back_end/service/EventService.java]: I'd use the repository here.

**Review by BrownMark — CHANGES_REQUESTED** [review]

File: web-app/src/utilities/AxiosInstance.js [hunk 9]
```diff
@@ -1,7 +1,8 @@
 import axios from 'axios';
 import moment from 'moment';
 
-const baseURL = process.env.DOMAIN;
+//const baseURL = process.env.DOMAIN;
+const baseURL = 'http://localhost:8081';
```
Comment (BrownMark) [inline web-app/src/utilities/AxiosInstance.js]: This needs removed and reverted.

File: BackEnd/src/main/java/com/unosquare/admin_core/back_end/configuration/mappings/presentation/CancelHolidayMappings.java [hunk 10]
```diff
@@ -23,8 +27,14 @@ protected void configure() {
...
         return  new PropertyMap <CancelHolidayViewModel, EventDTO>() {
             protected void configure() {
-                map().setEventId(source.getHolidayId());
+                EventTypesConverter eventTypesConverter = new EventTypesConverter();
+                EventStatusesConverter eventStatusConverter = new EventStatusesConverter();
 
+                using(eventTypesConverter.MapFromSourceToTarget()).map(EventTypes.ANNUAL_LEAVE, destination.getEventTypeId());
+                using(eventStatusConverter.MapFromSourceToTarget()).map(EventStatuses.DENIED, destination.getEventStatusId());
+                map().setEventId(source.getHolidayId());
+                map().setLastModified(LocalDate.now());
```
Comment (paulbrownlee) [inline BackEnd/src/main/java/com/unosquare/admin_core/back_end/configuration/mappings/presentation/CancelHolidayMappings.java]: Changed to map()

**Review by BrownMark — APPROVED** [review]

**Review by BrownMark — APPROVED** [review]

**Pull request merged**